### ISSUE TITLE
rokid xcraft questions

### ISSUE THREAD

**Issue by nickthirdeye** [RokidGlass/glass2-docs#155]
1.  the dual cameras they show on xcraft are not a stereo camera configuration which means their slam will not be as accurate. Maybe we could have them make a change there, or get some clarification on how their slam works. It would be great to have an Xvisio camera in there.

2.  Digging deeper what I don't like is that it is based on a Rockchip processor, that does not appear to have a DSP (maybe it does and I just can't see it)

3. is their slam documentation for xcraft on website dont see it anywhere? and also instructions on porting an ar core app to xcraft? how easy is it to port a samsung s10 android app running on android 9.0 to xcraft?
4. does google mobile services (gms) any ar core apps work on xcraft?

**Comment by Machaojoy** [comment]
> 1. the dual cameras they show on xcraft are not a stereo camera configuration which means their slam will not be as accurate. Maybe we could have them make a change there, or get some clarification on how their slam works. It would be great to have an Xvisio camera in there.
> 2. Digging deeper what I don't like is that it is based on a Rockchip processor, that does not appear to have a DSP (maybe it does and I just can't see it)
> 3. is their slam documentation for xcraft on website dont see it anywhere? and also instructions on porting an ar core app to xcraft? how easy is it to port a samsung s10 android app running on android 9.0 to xcraft?
> 4. does google mobile services (gms) any ar core apps work on xcraft?

1. Yes，they are not a stereo camera configuration according to product spec. it is designed for VIO SLAM(Camera+IMU).
2. It is NOT based on RK processor.
3. SLAM SDK is not ready. Porting a android APP seems to be easy to X-Craft.
4. NOT support GMS on X-Craft.

**Comment by nickthirdeye** [comment]
"I was just reading the datasheet for their chip this morning. They don't have a DSP so they're going to struggle to get SLAM running without it I think.  The chip does have an extra USB2.0 port that we could connect a camera module to I guess.  We haven't seen anyone do single camera slam successfully in a headset yet. Nreal is dual, pacific dual, all Qualcomm solutions are dual, Oculus is 4 camera, xvisio dual.  Getting the depth information you need is not easy with a single camera. 
there is zero slam documentation on your website too for xcraft. can your team answer that? Sales and tech team saying will be hard to sell for higher price than hololens 2 unless the slam is tested."  and when will slam documentation for xcraft and air pro be uploaded?

**Comment by Machaojoy** [comment]
> "I was just reading the datasheet for their chip this morning. They don't have a DSP so they're going to struggle to get SLAM running without it I think. The chip does have an extra USB2.0 port that we could connect a camera module to I guess. We haven't seen anyone do single camera slam successfully in a headset yet. Nreal is dual, pacific dual, all Qualcomm solutions are dual, Oculus is 4 camera, xvisio dual. Getting the depth information you need is not easy with a single camera.
> there is zero slam documentation on your website too for xcraft. can your team answer that? Sales and tech team saying will be hard to sell for higher price than hololens 2 unless the slam is tested." and when will slam documentation for xcraft and air pro be uploaded?

1. According to your description, you should also be a professional developer. We are developing VIO SLAM for specific scenarios in specific industries due to the limitation of single camera.
2. Dual Camera and 4 Camera Is a solution, we just chose another solution, according to our own product definition.
3. DSP is not necessary. SoC does not support it.

**Comment by nickthirdeye** [comment]
Yeah Soc does not support it but they’re gonna need it.  we don’t have much confidence in this so assume that this headset won’t have slam until you see it work in real life.  do you have any video examples of your xcraft slam? and when will slam documentation be uploaded? - engineer

**Comment by Machaojoy** [comment]
> Yeah Soc does not support it but they’re gonna need it. we don’t have much confidence in this so assume that this headset won’t have slam until you see it work in real life. do you have any video examples of your xcraft slam? and when will slam documentation be uploaded? - engineer

Hello，nickthirdeye，
1. For VIO SLAM  CPU is enough.
2. SLAM DEMO is not ready,AS you know,SLAM is important,but not all.
We pay more attention to solving customers' actual problems，and SLAM is just one part.
